### PR TITLE
Operators Workshop slides

### PR DESCRIPTION
**Pull request by rossf7**

### CODE REVIEW

File: content/operators-workshop/00-Agenda.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# Agenda
+
+1. Operators Overview
+2. Custom Resource Definition
+3. Operator Task
+4. Exercise 1. Implement using REST API
+5. Kubernetes client-go
+6. Exercise 2. Implement using client-go
+7. OperatorKit
```
Comment (rossf7) [inline content/operators-workshop/00-Agenda.md]: I think Agenda slides are done. Also the easiest! ;)

File: content/operators-workshop/01-Operators.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# What is an Operator
+
+- An application specific controller that extends the Kubernetes API
+- Manages a CRD (Custom Resource Definition)
+- CRD stores state and the operator performs operational tasks
+- Pattern first proposed by CoreOS in November 2016
```
Comment (rossf7) [inline content/operators-workshop/01-Operators.md]: Maybe more explanation is needed here.

File: content/operators-workshop/01-Operators.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+---
+
+# Giant Swarm Operators
+
+- azure-operator
+- aws-operator
+- kvm-operator
+- cert-operator
+- draughtsman-operator
```
Comment (rossf7) [inline content/operators-workshop/01-Operators.md]: Idea is we'd talk about some common operators e.g. Rook and also our operators and how they interact with each other.

File: content/operators-workshop/03-Operator-Task.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Operator Task
+
+TODO
```
Comment (rossf7) [inline content/operators-workshop/03-Operator-Task.md]: Rest of the slides are just placeholders.

File: content/operators-workshop/02-CRDs.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+---
+
+# Custom Resource Definition
+
+- Added as beta in 1.7
+- Reimplemented based on lessons learned from TPRs
+- Namespace or cluster scoped
+- Better pluralization support
+- Shortname support
```
Comment (rossf7) [inline content/operators-workshop/02-CRDs.md]: Maybe more explanation is needed here. Uncertain.

File: content/operators-workshop/02-CRDs.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+```
+/apis/containerconf.de/v1/postgresqlconfigs
+```
+
+- Similar to the standard endpoints
+
+```
+/api/v1/pods
+```
```
Comment (kopiczko) [inline content/operators-workshop/02-CRDs.md]: I think we should show evolution of the API - core (no group) & named groups. And say CRD creates new endpoint in the named group.

The core group, often referred to as the legacy group, is at the REST path /api/v1 and uses apiVersion: v1.
Example: `/api/v1/pods`

The named groups are at REST path /apis/$GROUP_NAME/$VERSION, and use apiVersion: $GROUP_NAME/$VERSION (e.g. apiVersion: batch/v1).
Example: `/apis/apps/v1beta1/deployments`

File: content/operators-workshop/01-Operators.md (hunk 2 above)
Comment (rossf7) [reply content/operators-workshop/01-Operators.md]: Done

File: content/operators-workshop/02-CRDs.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+spec:
+  group: containerconf.de
+  version: v1
+  scope: Namespaced
+  names:
+    plural: postgresqlconfigs
+    singular: postgresqlconfig 
+    kind: PostgreSQLConfig
+    shortNames: []
+```
```
[the remark this answers is not in the document]
Comment (rossf7) [reply content/operators-workshop/02-CRDs.md]: I've added an example CRO. I also switched to the crontab example as I think its simpler for explaining. I haven't added a cluster scoped example but we can talk about the differences.

File: content/operators-workshop/02-CRDs.md (hunk 6 above)
Comment (rossf7) [reply content/operators-workshop/02-CRDs.md]: Good point I've updated to include this.

File: content/operators-workshop/03-Operator-Task.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<!-- .slide: data-background-image="/layout/img/city_skyline_buildings_2.svg" data-background-size="50% 50%" data-background-position="bottom" -->
+# Operator Task
+
+---
```
Comment (rossf7) [inline content/operators-workshop/03-Operator-Task.md]: This section is new.

**Pull request merged**